### PR TITLE
Fixed an issue where pasting long text and typing in the middle moved the cursor to the end.

### PR DESCRIPTION
**Pull request by Ahmed-Naguib93**
refs: [MBL-17838](https://instructure.atlassian.net/browse/MBL-17838)
affects: Student , Teacher
release note: Fixed issues after pasting long text in inbox messages.

Fixed an issue where pasting long text and typing in the middle moved the cursor to the end.

## Test Plane
1- Open new message.
2- Set long text.
3- Start type in the middle on the text and set new line will work fine.

## Screenshots

https://github.com/user-attachments/assets/5b18cbbc-c84d-4432-acb5-8f65e957de96

## Checklist
- [ ] Tested on phone
- [ ] Tested on tablet
- [ ] Tested on iOS 17
- [ ] Tested on iOS 18

[MBL-17838]: https://instructure.atlassian.net/browse/MBL-17838?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Teacher Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>One or more files are below the minimum test coverage 50%</td>
    </tr>
  </tbody>
</table>


#### Release Note: 
Moved the cursor to the end when pasting long text

#### Affected Apps: Student, Teacher

[MBL-17838](https://instructure.atlassian.net/browse/MBL-17838)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 91.12% | 91.12% | 0.01%
Core/Core/Planner/CalendarEvent/Model/API/PostCalendarEventRequest.swift | 0% | 0% | 0%
Core/Core/Planner/CalendarEvent/View/CustomFrequencyComponents/OccurrencesCountInputDialog.swift | 0% | 0% | 0%
Core/Core/Planner/CalendarEvent/Model/Helpers/RecurrenceRule+SelectionDescription.swift | 43.33% | 43.33% | 0%
Core/Core/Grades/View/CustomSwipeAction.swift | 0% | 0% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f5432f2f7cbb9838131043b385ebd9fbab089d79
</p>

**Review by vargaat — CHANGES_REQUESTED** [review]
I get this strange behavior on iOS 17.5.1. Besides that, I think we can't just scroll to the end of the text (where attachments are) because the user is not necessarily pasting text at the end of the message. The original bug was about when you enter some characters after pasting a long text the characters are not entered to where the cursor is visible but instead at the end of the document. If we could fix that alone would be enough, if we could scroll to the end of the **pasted** text that would nice to have.

https://github.com/user-attachments/assets/9e1fcf39-6b6b-4d43-b5fb-d6171ea8a3c8

**Comment by Ahmed-Naguib93** [comment]
> Moved the cursor to the end when pasting long text

Thanks, i fixed the issues please check again 

https://github.com/user-attachments/assets/28aa7053-8ad2-462b-b280-27cb0c95b1ab

File: Core/Core/Inbox/ComposeMessage/View/UITextViewWrapper.swift [hunk 1]
```diff
@@ -1,54 +0,0 @@
...
-//
-// You should have received a copy of the GNU Affero General Public License
-// along with this program.  If not, see <https://www.gnu.org/licenses/>.
-//
-
-import Foundation
-import SwiftUI
-
-struct UITextViewWrapper: UIViewRepresentable {
-    @Binding var text: String
```
Comment (Ahmed-Naguib93) [inline Core/Core/Inbox/ComposeMessage/View/UITextViewWrapper.swift:23]: I removed this file, because no longer used

**Comment by Ahmed-Naguib93** [comment]
> `InstUI.TextEditorCell` on iOS 17.

Yes, I have tested it on iOS 18.0.1, and it is working fine. This is a standard UITextView, an older controller.

**Comment by rh12** [comment]
> Yes, I have tested it on iOS 18.0.1, and it is working fine. This is a standard UITextView, an older controller.

Sorry, I tested the right thing, but wrote about the other thing... anyway, please disregard the PR comment above about the doc comment in code.

Approving

**Review by rh12 — APPROVED** [review]

**Review by szabinst — APPROVED** [review]
QA + 1

**Review by suhaibabsi-inst — APPROVED** [review]
QA + 1

**Review by vargaat — APPROVED** [review]

**Pull request merged**